src/app/restaurant/restaurant.component.css

```diff
@@ -10,3 +10,25 @@
   font-size: 16px;
   color: #333;
 }
+
+app-star-icon {
+  cursor: pointer;
+}
+
+.active {
+  color: gold;
```

Comment:
Another solution could be utilising svg's fill property which would completely fill the star with color